### DISCUSSION TITLE
Continued support for --learn-lib?

### DISCUSSION THREAD

**Discussion by enonimos** [vdemichev/DiaNN#1244] [Q&A] [answered]
Hello,
In the previous DIA-NN 1.8.2, I had used the --learn-lib command along with several other commands (see below) to use experimental spectral library with PTM to train deep learning predictor and use that to generate an in silico library.  My search results using this strategy was superior than using default model.  I see that the --learn-lib and --strip-unknown-mods are no longer listed in the documentation and when I try to use the commands on version 1.9.2, --monitor-mod and --strip-unknown-mods are not recognized, and overall based on the output, it doesn't appear to execute the learn-lib function.  Can anyone confirm if this can be still be done in 1.9.2 or not?
Thank you in advance,
Todd
--var-mod UniMod:378,72.0211,K --monitor-mod UniMod:378 --strip-unknown-mods --no-cut-after-mod UniMod:378

**Comment by vdemichev** [comment, marked as answer]
Hi Todd,

This is no longer supported. The reason is basically that there are no known cases when this 'basic' predictor outperformed the deep learning one, hence the reason to discontinue. If the deep learning one really struggles with a particular modification, the solution is to use e.g. AlphaPeptDeep to generate a library for DIA-NN.

--strip-unknown-mods is now default in DIA-NN since 1.9, e.g. no need to turn it on.

Best,
Vadim

**Reply by enonimos** [reply]
Thank you Vadim, that is very curious.  My n=1 case then is the exception.  Though I agree, it was a modest benefit and I don't know if the benefit would reproduce in other training datasets.  The AlphaPeptDeep tool seems like a great option, thank you again.  
Best,
Todd

**Reply by enonimos** [reply]
Hello again Vadim, I had one follow-up question about your suggestion to use AlphaPeptDeep.  When generating a predicted library using this tool, the output formats are either hdf or tsv. Given the size of the predicted library, I would like to convert it to speclib. I loaded the tsv into DIANN 1.9.2 without "Generate spectral library" enabled as this would generate a parquet.  However, the output I get is the *.skyline.speclib file.  I recall reading that the "skyline" generated file should only be used for the purposes of skyline analysis.  Can you confirm if this is correct and/or suggest how to generate a properly formatted speclib from an in silico predicted library in tsv format.
Thanks,
Todd

**Reply by vdemichev** [reply]
Hi Todd,

>  I recall reading that the "skyline" generated file should only be used for the purposes of skyline analysis

In practice will work fine as input for DIA-NN, it's just we don't 'support' it in the sense that this functionality is not tested and can stop working in the future versions.

Best
Vadim
